python-avd/pyavd/_eos_designs/structured_config/network_services/router_ospf.py

```diff
@@ -20,85 +23,90 @@ class RouterOspfMixin(Protocol):
-
-                if vrf.ospf.redistribute_connected.enabled:
-                    process_redistribute["connected"] = {"enabled": True}
-                    if route_map := vrf.ospf.redistribute_connected.route_map:
-                        process_redistribute["connected"]["route_map"] = route_map
-
-                process["redistribute"] = process_redistribute or None
-
-                # Strip None values from process before adding to list
-                process = {key: value for key, value in process.items() if value is not None}
-
-                append_if_not_duplicate(
-                    list_of_dicts=ospf_processes,
-                    primary_key="id",
-                    new_dict=process,
-                    context="OSPF Processes defined under network services",
-                    context_keys=["id"],
-                )
-
+                process._update(id=process_id, passive_interface_default=True)
+
+                if vrf.name != "default":
+                    process.vrf = vrf.name
+                if vrf_router_id := self.get_vrf_router_id(vrf, tenant, vrf.ospf.router_id):
+                    process.router_id = vrf_router_id
+                if vrf.ospf.bfd:
+                    process.bfd_enable = vrf.ospf.bfd
+                process.max_lsa = vrf.ospf.max_lsa
+                self._update_ospf_redistribute(process, vrf)
+
+                self.structured_config.router_ospf.process_ids.append(process)
         # If we have static_routes in default VRF and not EPVN, and underlay is OSPF
         # Then add redistribute static to the underlay OSPF process.
         if self._vrf_default_ipv4_static_routes["redistribute_in_underlay"] and self.shared_utils.underlay_routing_protocol in ["ospf", "ospf-ldp"]:
-            ospf_processes.append({"id": self.inputs.underlay_ospf_process_id, "redistribute": {"static": {"enabled": True}}})
-        if ospf_processes:
-            return {"process_ids": ospf_processes}
+            self.structured_config.router_ospf.process_ids.obtain(self.inputs.underlay_ospf_process_id).redistribute.static.enabled = True
 
-        return None
+    def _update_ospf_redistribute(
+        self, process: EosCliConfigGen.RouterOspf.ProcessIdsItem, vrf: EosDesigns._DynamicKeys.DynamicNetworkServicesItem.NetworkServicesItem.VrfsItem
+    ) -> None:
+        """
+        Configures OSPF route redistribution settings for the given VRF.
+
+        This method enables redistribution of BGP and connected routes into OSPF,
+        setting the associated route maps if specified.
+
+        Args:
+            process: The OSPF process configuration object.
+            vrf: The VRF object containing OSPF redistribution settings.
+        """
+        if vrf.ospf.redistribute_bgp.enabled:
+            process.redistribute.bgp.enabled = True
+            if route_map := vrf.ospf.redistribute_bgp.route_map:
+                process.redistribute.bgp.route_map = route_map
```

Comment:
This could possibly be done with a _cast_as since the keys are 1:1. If you do that, it ends up with two lines and no "if` so it would be easier done directly in the process creation so remove this function.